kalamine/utils.py

```diff
@@ -35,8 +43,30 @@ class Layer(IntEnum):
     ALTGR = 4
     ALTGR_SHIFT = 5
 
+    def next(self) -> 'Layer':
+        """The next layer in the layer ordering."""
+        return Layer(int(self)+1)
+
+    def necromance(self) -> 'Layer':
+        """Remove the effect of the dead key if any."""
+        if self == Layer.ODK:
+            return Layer.BASE
+        elif self == Layer.ODK_SHIFT:
+            return Layer.SHIFT
+        return self
+
+@dataclass
+class DeadKeyDescr:
+    char: str
+    name: str
+    base: str
+    alt: str
+    alt_space: str
+    alt_self: str
```

Comment:
Correct. No idea if this is doable with Python, but `base` and `alt` are actually char arrays rather than strings : they both contain exactly N characters and `alt[i]` is the altered version of `base[i]` for any `i` between zero and N-1.

Reply:
Weirdly enough, Python does not have a char type. I never learned why. So we could have `List[str]` here but not `List[char]`.